pulsar-broker/src/test/java/com/yahoo/pulsar/websocket/proxy/ProxyPublishConsumeTest.java

```diff
@@ -101,12 +104,20 @@ public void socketTest() throws Exception {
             Assert.assertTrue(produceSocket.getBuffer().size() > 0);
             Assert.assertEquals(produceSocket.getBuffer(), consumeSocket.getBuffer());
         } finally {
-            try {
-                consumeClient.stop();
-                produceClient.stop();
-            } catch (Exception e) {
-                log.error(e.getMessage());
-            }
+            ExecutorService executor = newFixedThreadPool(1);
+            executor.submit(() -> {
+                try {
+                    consumeClient.stop();
+                    produceClient.stop();
+                    log.info("proxy clients are stopped successfully");
+                } catch (Exception e) {
+                    Thread.currentThread().interrupt();
+                    log.error(e.getMessage());
+                }
+            });
+            // let's wait for clients to be stopped
+            Thread.sleep(2000);
```

Comment:
Instead of always sleeping 2 seconds, we could use the future returned by `executor.submit()` and do a `get()`  with 2sec timeout. 

Also, can you do the same change in `ProxyPublishConsumeWithoutZKTest`  and `ProxyPublishConsumeTls` ?